Automattic/woocommerce-payments
# Ask for login when buying a subscription with WooPay

Fixes #4435

#### Changes proposed in this Pull Request









* Show a login notice before checkout when a existing logged out user buys a subscription and WooPay is enabled.

#### Testing instructions





* Enable subscriptions for WooPay on merchant site.
* Buy a subscription on merchant site.
* Access My Account > Logout.
* Buy another subscription.
* Type a valid WooPay e-mail.
* Instead of showing the SMS OTP, it should show the same notice when you checkout a subscription on merchant site while having a merchant site account, asking the user to login (notice that the "Please login" link on that notice is a `#` and don't redirect you to the `My account` page even when you checkout on merchant site with a existing account).
* Do the same flow while logged in or without subscriptions on the cart, it should show the SMS OTP normally.

-------------------


**Post merge**
